src/abcdmicro/build_template.py

```diff
@@ -0,0 +1,428 @@
+                error_message = f"Input volume dimensions must be 2D or 3D. Found {volume.get_array().ndim}D instead."
+                raise ValueError(error_message)
+            reformatted[modality].append(volume)
+    return dict(reformatted)
+
+
+def build_multi_metric_template(
+    subject_list: Sequence[Mapping[str, VolumeResource]],
+    initial_template: dict[str, VolumeResource] | None = None,
+    weights: dict[str, np.floating] | None = None,
+    iterations: int = 3,
+) -> dict[str, InMemoryVolumeResource]:
+    """
+    Constructs an unbiased mean shape template from a list of input volumes using
+    an iterative group-wise registration approach based on ANTs, utilizing
+    multiple image modalities simultaneously. See 'build_template' for more details.
+
+    NOTE: The current implementation assumes input images are roughly pre-aligned.
+
+    Args:
+        subject_list: A list of input data, where each subject is a dictionary mapping modality names (string) to their corresponding
+        3D image volumes (VolumeResource objects).
+        initial_template: An optional starting template volume. If None, the
+            initial template is the simple average of all input volumes.
+        weights: The weight given to each volume type during the multivariate registration step. If None, equal weights are assumed.
+            The length must match the number of volumes specified per subject.
+        iterations: The number of iterations for the template refinement process.
+    Returns:
+        A VolumeResource object representing the final group-wise mean template (per metric)
+    """
+
+    # Get the list of modalities from the first subject
+    modalities = list(subject_list[0].keys())
+    primary_mod = modalities[0]
+
+    # Check the the input variables all reference the same image modalities
+    for d, name in zip(
+        [weights, initial_template], ["weights", "initial_template"], strict=False
+    ):
+        if d and list(d.keys()) != modalities:
+            error_msg = (
+                f"Inconsistent keys detected in '{name}' compared to subject_list"
+            )
+            raise ValueError(error_msg)
+
+    # Convert to a list of volumes for each metric type
+    volume_list = _reformat_subject_list(subject_list)
+
+    # Initialize the template - dictionary of volume resources
+    if initial_template is None:
+        initial_template = {}
+        for m in modalities:
+            initial_template[m] = average_volumes(volume_list[m])
+
+    current_template = {}
+    for m in modalities:
+        current_template[m] = ants.from_numpy(initial_template[m].get_array())
+
+    # weights for multivariate extras are relative to the primary metric
+    if weights is None:
+        weights = dict.fromkeys(modalities, 1.0)  # Equal weighting
+
+    weights = {m: weights[m] / weights[primary_mod] for m in modalities}
+
+    n_subj = len(subject_list)
+    for _i in range(iterations):
```

Comment:
```suggestion
    for _ in range(iterations):
```

(more clearly communicates that we do not use `_i`)